### ISSUE TITLE
Report error when variable is used uninitialized in a particular scope

### ISSUE THREAD

**Issue by mthuurne** [python/mypy#7116]
* Are you reporting a bug, or opening a feature request?
Feature request.

The following code does not trigger any warnings in mypy 0.711:

```py
def f(b: bool) -> int:
    if b:
        x = 123
        return x
    else:
        return x
```

I'm not sure 'uninitialized' qualifies as a type, but mypy does track execution paths to narrow types of variables within certain scopes, so it may not be too hard to detect errors like in the example above.

**Comment by msullivan** [comment]
Yeah, this would be useful. Duplicate of #4019, I think.
